Lib/socketserver.py

```diff
@@ -249,6 +246,7 @@ def shutdown(self):
         deadlock.
         """
         self.__shutdown_request = True
+        self.server_shutdown() # selector.select() will return immediately
```

Comment:
Please follow the comment style of this file. Comments are written before the comment line.

Reply:
Fixed